lib/manageiq/api/common/request.rb

```diff
@@ -31,17 +44,24 @@ def self.with_request(request)
         end
 
         def self.current_forwardable
-          raise ManageIQ::API::Common::HeadersNotSet, "Current headers have not been set" unless current
-          current.forwardable
+          current!.forwardable
         end
 
         attr_reader :headers, :original_url
 
-        def initialize(headers:, original_url:, **kwargs)
+        def initialize(headers:, original_url:, **_kwargs)
           headers = from_hash(headers) if headers.kind_of?(Hash)
           @headers, @original_url = headers, original_url
         end
 
+        def request_id
+          headers[REQUEST_ID_KEY]
+        end
+
+        def identity
+          JSON.parse(Base64.decode64(headers[IDENTITY_KEY]))
```

Comment:
why not cached the identity here rather than in each user instance?
should raise an error if headers does not have key `IDENTITY_KEY`.

Reply:
I like the idea of caching it here.  I'll do that.

If the identity key is missing, should this raise or just return nil.  I was thinking just return nil, but I'm not sure.

Reply:
From an RBAC perspective if the user doesn't exist nothing can happen.  So maybe we should `raise` instead of return `nil`.